sklearn/base.py

```diff
@@ -61,10 +61,14 @@ def clone(estimator, safe=True):
         if not safe:
             return copy.deepcopy(estimator)
         else:
-            raise TypeError("Cannot clone object '%s' (type %s): "
-                            "it does not seem to be a scikit-learn estimator "
-                            "as it does not implement a 'get_params' methods."
-                            % (repr(estimator), type(estimator)))
+            if issubclass(estimator, BaseEstimator):
+                raise TypeError("You should provide an instance of " +
+                                "scikit-learn estimator instead of a class.")
+            else:
+                raise TypeError("Cannot clone object '%s': "
+                                "it is a class rather than an instance."
+                                % (repr(estimator)))
```

Comment:
these 2 errors  message say more or less thing. I think you should change the condition

`issubclass(estimator, BaseEstimator):` to `isinstance(estimator, type):` and put back the previous error message inside of the `else:` clause

Reply:
Thanks I changed to this

Reply:
Sorry if I'm misunderstanding but this still seems to error

Reply:
the second error message should be about not implementing a get_params method, like it was before

Reply:
Ah I understand now apologies